### ISSUE TITLE
Filters / Actions not working for ACF in WooCommerce >= 5.5.x

### ISSUE THREAD

**Issue by R3dRidle** [woocommerce/woocommerce#30310]
**Prerequisites (mark completed items with an [x]):**
- [X] I have have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
The bug is since WooCommerce 5.5.* 
I have WooCommerce installed and ACF. 
When you visit the homepage you see contact data. https://www.gebuwin0.int.webton.dev/
When searching for a product, you see that the contact data is still there, but it is not formatted: https://www.gebuwin0.int.webton.dev/?post_type=product&s=test123

**Expected behavior**
I expected to see formatted contact data on https://www.gebuwin0.int.webton.dev/?post_type=product&s=test123
In WooCommerce 5.4.2 this works without problems.

**Actual behavior**
On https://www.gebuwin0.int.webton.dev/?post_type=product&s=test123 there is no formatting of the data. It's almost like no filters / actions are run.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:
1. Install WooCommerce 5.5.*
2. Install ACF
3. Create ACF WYSIWYG field (and save some data in it)
4. Add ACF field in the template
5. See the formatting not working on the search page ?post_type=product&s=test123.

**Isolating the problem (mark completed items with an [x]):**
- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://www.gebuwin0.int.webton.dev
Site address (URL): https://www.gebuwin0.int.webton.dev
WC Version: 5.5.1
REST API Version: ✔ 5.5.1
WC Blocks Version: ✔ 5.3.2
Action Scheduler Version: ✔ 3.2.1
WC Admin Version: ✔ 2.4.1
Log Directory Writable: ✔
WP Version: 5.7.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2
PHP Version: 7.4.8
PHP Post Max Size: 256 MB
PHP Time Limit: 300
PHP Max Input Vars: 5000
cURL Version: 7.29.0
NSS/3.53.1

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.23-MariaDB
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.5.1
WC Database Prefix: fiby_
Totale databasegrootte: 63.20MB
Database datagrootte: 42.32MB
Database index grootte: 20.88MB
fiby_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
fiby_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_woocommerce_order_itemmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
fiby_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
fiby_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_actionscheduler_actions: Data: 0.06MB + Index: 0.13MB + Engine InnoDB
fiby_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_actionscheduler_logs: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
fiby_alm: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_alm_unlimited: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_commentmeta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
fiby_comments: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
fiby_gf_addon_feed: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_gf_draft_submissions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_gf_entry: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
fiby_gf_entry_meta: Data: 0.08MB + Index: 0.13MB + Engine InnoDB
fiby_gf_entry_notes: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
fiby_gf_form: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_gf_form_meta: Data: 0.20MB + Index: 0.00MB + Engine InnoDB
fiby_gf_form_revisions: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_gf_form_view: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
fiby_gf_rest_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_icl_content_status: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_icl_core_status: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_icl_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_icl_languages: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_icl_languages_translations: Data: 0.20MB + Index: 0.17MB + Engine InnoDB
fiby_icl_locale_map: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_icl_message_status: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_icl_mo_files_domains: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
fiby_icl_node: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_icl_reminders: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_icl_strings: Data: 5.52MB + Index: 11.05MB + Engine InnoDB
fiby_icl_string_batches: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_icl_string_packages: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_icl_string_pages: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_icl_string_positions: Data: 1.52MB + Index: 0.25MB + Engine InnoDB
fiby_icl_string_status: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_icl_string_translations: Data: 1.52MB + Index: 1.78MB + Engine InnoDB
fiby_icl_string_urls: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_icl_translate: Data: 1.52MB + Index: 0.27MB + Engine InnoDB
fiby_icl_translate_job: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
fiby_icl_translations: Data: 0.22MB + Index: 0.58MB + Engine InnoDB
fiby_icl_translation_batches: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_icl_translation_downloads: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_icl_translation_status: Data: 5.52MB + Index: 0.06MB + Engine InnoDB
fiby_imagify_files: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
fiby_imagify_folders: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_links: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
fiby_options: Data: 4.56MB + Index: 0.17MB + Engine MyISAM
fiby_postmeta: Data: 17.13MB + Index: 2.85MB + Engine MyISAM
fiby_posts: Data: 0.89MB + Index: 0.28MB + Engine MyISAM
fiby_redirection_404: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
fiby_redirection_groups: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_redirection_items: Data: 0.06MB + Index: 0.09MB + Engine InnoDB
fiby_redirection_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
fiby_signups: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
fiby_termmeta: Data: 0.09MB + Index: 0.13MB + Engine InnoDB
fiby_terms: Data: 0.01MB + Index: 0.04MB + Engine MyISAM
fiby_term_relationships: Data: 0.36MB + Index: 0.72MB + Engine MyISAM
fiby_term_taxonomy: Data: 0.03MB + Index: 0.03MB + Engine MyISAM
fiby_usermeta: Data: 0.04MB + Index: 0.02MB + Engine MyISAM
fiby_users: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
fiby_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
fiby_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
fiby_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
fiby_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_wc_product_meta_lookup: Data: 0.06MB + Index: 0.09MB + Engine InnoDB
fiby_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_wpfront_ure_login_redirect: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_wpfront_ure_options: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_wpr_rucss_resources: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
fiby_wpr_rucss_used_css: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
fiby_yith_wapo_addons: Data: 0.28MB + Index: 0.00MB + Engine InnoDB
fiby_yith_wapo_blocks: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
fiby_yith_wapo_groups: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_yith_wapo_types: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
fiby_yith_wcan_filter_sessions: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
fiby_yith_wccl_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fiby_yoast_indexable: Data: 0.39MB + Index: 0.28MB + Engine InnoDB
fiby_yoast_indexable_hierarchy: Data: 0.06MB + Index: 0.05MB + Engine InnoDB
fiby_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fiby_yoast_primary_term: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
fiby_yoast_seo_links: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fiby_yoast_seo_meta: Data: 0.08MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

acf-field: 272
acf-field-group: 46
attachment: 1767
download: 2
nav_menu_item: 57
oembed_cache: 15
page: 72
post: 56
product: 293
revision: 820
shop_order: 15
yith_wc_qof_1: 3
yith_wcan_preset: 5
ywctm-button-label: 4
ywdpd_discount: 3

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Foutmeldingen zouden niet getoond moeten worden aan bezoekers.

### Active Plugins (4) ###

Advanced Custom Fields PRO: door Delicious Brains – 5.9.8
Font Awesome: door Font Awesome – 4.0.0-rc23
WooCommerce: door Automattic – 5.5.1
WP Rollback: door Impress.org – 1.7.1

### Inactive Plugins (44) ###

ACF: Better Search: door Mateusz Gbiorczyk – 3.8.1
ACF Content Analysis for Yoast SEO: door Thomas Kräftner
ViktorFroberg
marol87
pekz0r
angrycreative
Team Yoast – 3.0.1

Advanced Custom Fields: Image Crop Add-on: door Anders Thorborg – 1.4.12
Advanced Custom Fields Multilingual: door OnTheGoSystems – 1.8.2
Ajax Load More: door Darren Cooney – 5.5.0
Ajax Load More: Pro: door Darren Cooney – 1.2.7
Akismet Anti-Spam: door Automattic – 4.1.10
Category Order and Taxonomy Terms Order: door Nsp-Code – 1.5.7.5
Classic Editor: door WordPress Contributors – 1.6
Custom Post Type UI: door WebDevStudios – 1.9.2
Favicon by RealFaviconGenerator: door Philippe Bernard – 1.3.20
Force Regenerate Thumbnails: door Pedro Elsner – 2.0.6
GP Multi-page Navigation: door Gravity Wiz – 1.1.2
Gravity Forms: door Gravity Forms – 2.5.7
Gravity Forms Multilingual: door OnTheGoSystems – 1.6.0
Gravity Perks: door Gravity Wiz – 2.2.5
Imagify: door WP Media – 1.10
ManageWP - Worker: door GoDaddy – 4.9.8
Members: door MemberPress – 3.1.5
Redirection: door John Godley – 5.1.2
Webton Extended functions: door Tieme Alberts & Dennis de Groot (Webton B.V.) – 1.0.38
Webton Laposta Functions: door Tieme Alberts (Webton B.V.) – 1.0.4
Webton Media Categories: door Dennis de Groot – 1.0.0
WooCommerce Multilingual: door OnTheGoSystems – 4.11.6
WordPress Force HTTPS: door A. Kai Armstrong – 0.1.4
WP Add Mime Types: door Kimiya Kitani – 3.0.0
WP Mail SMTP: door WPForms – 2.9.0
WPML Media: door OnTheGoSystems – 2.6.4
WPML Multilingual CMS: door OnTheGoSystems – 4.4.10
WPML String Translation: door OnTheGoSystems – 3.1.8
WPML Translation Management: door OnTheGoSystems – 2.10.6
WP Realtime Sitemap: door Dennis de Groot – 2.0
WP Rocket: door WP Media – 3.9.0.5
YITH Quick Order Forms for WooCommerce Premium: door YITH – 1.2.18
YITH WooCommerce Added to Cart Popup Premium: door YITH – 1.8.4
YITH WooCommerce Ajax Product Filter Premium: door YITH – 4.1.0
YITH WooCommerce Ajax Search Premium: door YITH – 1.8.6
YITH WooCommerce Catalog Mode Premium: door YITH – 2.1.4
YITH WooCommerce Compare Premium: door YITH – 2.5.3
YITH WooCommerce Dynamic Pricing and Discounts Premium: door YITH – 2.1.10
YITH WooCommerce Product Add-ons & Extra Options Premium: door YITH – 2.0.1
YITH WooCommerce Request A Quote Premium: door YITH – 3.1.6
Yoast Duplicate Post: door Enrico Battocchi & Team Yoast – 4.1.2
Yoast SEO: door Team Yoast – 16.7

### Dropin Plugins (1) ###

advanced-cache.php: advanced-cache.php

### Must Use Plugins (2) ###

Webton_mu_plugin.php: door  –
yith_debugger.php: door Armando Liccardo (YITH) – 1.0.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: composite (composite)
external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Winkelbasis: #1756 - /products/
Winkelmand: #1754 - /shopping-cart/
Afrekenen: #1748 - /checkout/
Mijn account: #1752 - /my-account/
Algemene voorwaarden: ❌ Pagina niet ingesteld

### Theme ###

Name: Twenty Twenty-One
Version: 1.0 (update naar versie 1.3 is beschikbaar)
Author URL: https://nl.wordpress.org/
Child Theme: ❌ – Als je wijzigingen aanbrengt in WooCommerce of in een hoofdthema dat je niet zelf gebouwd hebt
raden we het gebruik van een child thema aan. Zie: Hoe maak je een child thema

WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Afgerond: 98
Oldest: 2021-06-24 07:31:30 +0000
Newest: 2021-07-19 12:03:09 +0000


### Status report information ###

Generated at: 2021-07-19 14:39:35 +02:00
`
```
</details>

**Comment by tammullen** [comment]
Hi @R3dRidle 

While our goal is to address all the issues reported in this repository, GitHub should be treated as a place to report WooCommerce bugs only. 
​
The issue you are having seems to be caused by the 3rd party plugin Advanced Custom Fields. 
​
Please contact the author of the plugin directly for further help.

**Comment by R3dRidle** [comment]
@tammullen I am not too sure about that. The issue occors only after update to 5.5.x.
Also they (ACF) sent me here for the bug.

Now I don't know where to go. You say to go to ACF, ACF says go to WooCommerce....

**Comment by tammullen** [comment]
Thanks @R3dRidle for the additional information.

Can you provide any other details from ACF on the filters or actions they believe there is an issue with?

**Comment by R3dRidle** [comment]
@tammullen This is what they said on the matter. So I realy don't know.

Thanks for reaching out to us and for the follow-up. 

Unfortunately, the issue seems hooked to the WooCommerce update and we are not able to assist as much. 

I would recommend that you check with the WooCommerce team for a possible workaround for the same. Hopefully, they will be able to shed some light on the same. 

Let me know how it goes.

**Comment by tammullen** [comment]
Thanks @R3dRidle for their response.

I'm not sure there is enough information for them to evaluate it but this requires further feedback from the WooCommerce Core team. I am adding the `needs developer feedback` label to this issue so that the Core team could take a look. 
​
Please note it may take a few days for them to get to this issue. Thank you for your patience.

**Comment by tammullen** [comment]
Maybe related: https://github.com/woocommerce/woocommerce/issues/30267

**Comment by lgladdy** [comment]
Hey everyone, joining from the ACF dev team here 👋

We've been able to reproduce this issue. It only affects the WooCommerce powered search results page. It's caused by the same issue as #30298 - WooCommerce is modifying more than just the main query on the search results page.

**Comment by tammullen** [comment]
Closing as this will hopefully be covered by the fix in https://github.com/woocommerce/woocommerce/pull/30335 which is now available in 5.5.2 released today.
